apps/authentication-service/src/shared/auth/authService.ts

```diff
@@ -0,0 +1,112 @@
+import fs from "fs";
+import path from "path";
+import os from "os";
+import { TokenSet } from "auth0";
+import { decode, JwtPayload } from "jsonwebtoken";
+
+import envs from "@/config";
+import { IAuthService, ICredentialRequestBody } from "@/shared/auth/authModel";
+import { ISingleton } from "@/shared/models";
+
+export const authEndpoint = "oauth/token";
+
+export const AuthService: ISingleton<IAuthService> = (() => {
```

Comment:
Why is this a function that immediately calls itself to return an object? Why isn't it just an object?

Reply:
Going beyond this I don't understand why this needs to be something that gets instantiated.

Why don't we just export getAccessToken directly?

Reply:
This is a pattern that is used for Singletons, instead of using a class. The token is stored in memory during the lifetime of the service. I want the same instance to be called when any other code needs the token. I found this pattern in multiple examples of how to create a singleton without using a class.

The getInstance returns the single instance, unless `instance` isn't already set. 

I'm open to cleaner ways to do this.

Reply:
If I were to write this file I would write it like this:

`let token: string;

const setCachedToken = () => {}
const getCachecToken = () => {}
const getIsTokenExpired = () => {}
export const getAccessToken = () => {}`

Anywhere that needed get access token would just call it directly
`
import { getAccessToken } from "../"

await getAccessToken()
`

As far as I understand the only thing any other file cares about from this file is the getAccessToken function.

Do you see a benefit in using the singleton pattern over just exporting the function that is needed?

Reply:
I've simplified it. I wasn't sure if something that simple would persist the token value, but it works. Thanks for the suggestion.